xpass/xpass.cc

```diff
@@ -211,6 +222,12 @@ void XPassAgent::recv_data(Packet *pkt) {
   update_rtt(pkt);
 }
 
+void XPassAgent::recv_nack(Packet *pkt) {
+  hdr_tcp *tcph = hdr_tcp::access(pkt);
+  // set t_seqno_ for retransmission
+  t_seqno_ = tcph->ackno();
+}
+
```

Comment:
You don't need to consider XPASS_RECV_NSTATE. This is just the number of the total state.